configure.ac

```diff
@@ -0,0 +1,80 @@
+## -*- mode: autoconf; autoconf-indentation: 4; -*-
+##
+##  nloptr configure.ac
+##
+##  nloptr -- R interface to NLopt
+##
+##  Copyright (C) 2014-2022  Dirk Eddelbuettel
+##
+## This file is licensed under the GPL-2 or later just like most of my
+## Open Source code, and is granted an exemption (should it be needed)
+## for inclusion into nloptr
+
+# require at least autoconf 2.61
+AC_PREREQ([2.69])
+AC_INIT([nloptr],[2.0.0])
+
+## Set R_HOME, respecting an environment variable if set
+: ${R_HOME=$(R RHOME)}
+if test -z "${R_HOME}"; then
+   AC_MSG_ERROR([Could not determine R_HOME.])
+fi
+: ${R_ARCH=${R_ARCH}}
+
+CXX=`"${R_HOME}/bin/R" CMD config CXX`
+if test -z "${CXX}"; then
+    AC_MSG_ERROR([No C++ compiler available])
+fi
+CXXFLAGS=`"${R_HOME}/bin/R" CMD config CXXFLAGS`
+CPPFLAGS=`"${R_HOME}/bin/R" CMD config CPPFLAGS`
+CXX11FLAGS=`"${R_HOME}/bin/R" CMD config CXX11FLAGS`
+CXX14FLAGS=`"${R_HOME}/bin/R" CMD config CXX14FLAGS`
+AC_LANG(C++)
+AC_REQUIRE_CPP
+AC_PROG_CXX
+
+## Default to build from source
+need_to_build="yes"
+
+## But: Can we use pkg-config?
+AC_PATH_PROG(have_pkg_config, pkg-config, no)
+## If yes, also check for whether pkg-config knows nlopt
+if test x"${have_pkg_config}" != x"no"; then
+    AC_MSG_CHECKING([if pkg-config knows NLopt])
+    if pkg-config --exists nlopt; then
+        AC_MSG_RESULT([yes])
+        nlopt_include=$(pkg-config --cflags nlopt)
+        nlopt_libs=$(pkg-config --libs nlopt)
+        AC_SUBST([NLOPT_INCLUDE], "${nlopt_include}")
+        AC_SUBST([NLOPT_LIBS],    "${nlopt_libs}")
+    else
+        AC_MSG_RESULT([no])
+        have_pkg_config="no"
+    fi
+fi
+
+## And if we have pkg-config, use it to test minimal version
+if test x"${have_pkg_config}" != x"no"; then
+    AC_MSG_CHECKING([for pkg-config checking NLopt version])
+    if pkg-config --atleast-version=2.7.0 nlopt; then
+        AC_MSG_RESULT([>= 2.7.0])
+        need_to_build="no"
+    else
+        AC_MSG_RESULT([insufficient])
+        AC_MSG_ERROR([NLopt 2.7.0 or later is required.])
+    fi
+fi
+
+## So do we need to build
+if test x"${need_to_build}" != x"no"; then
+   AC_MSG_RESULT([using NLopt via local build...])
+   tools/cmake_call.sh
+   nlopt_include=""
+   nlopt_libs="-Lnlopt${R_ARCH}/lib -lnlopt"
```

Comment:
I re-ran the GH action script `R CMD check` and everything works except on Windows.
I suspect this is due to the way you auto-generate the `Makevars` file in `configure.ac` in this line.
The generated `Makevars` should in the end look like exactly as the one you suppressed (since it will be auto-generated by `Makevars.in` now).
So I think here it should be `nlopt_libs="-Lnlopt\$(R_ARCH)/lib -lnlopt"` instead of `nlopt_libs="-Lnlopt${R_ARCH}/lib -lnlopt"`.

Reply:
No, not usually.

Windows uses src/Makevars.win.  We can just create one such file, fixed content.  Over one thousands other CRAN packages do:  https://github.com/search?q=org%3Acran+src%2FMakevars.win  So if you want I could copy to _previous_ and _constant_ `src/Makevars` in as `src/Makevars.win` (or you do after merge).

Also, in the comment the suggested difference is hard to read.  Are you suggesting to add a backslash? Why?

Reply:
Also, note that _if and when one had system nlopt_ the `src/Makevars` looks different.  Which is the point after all:

```sh
edd@rob:~/git/nloptr(feature/build_options)$ cat src/Makevars
CXX_STD = CXX11
PKG_CPPFLAGS = -I../inst/include 
PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) -lnlopt
edd@rob:~/git/nloptr(feature/build_options)$ 
```

Reply:
I just made that change of restoring the previous as `src/Makevars.win`.  As `configure.win` is also unchanged, we should be as good as we were before _on Windows_ and now have the optional use of a NLopt installation.  With relatively minimal changes.  Hope that suits.

Reply:
Thanks. I just meant to avoid using a `Makevars.win` because I think that if no such file is there, then Windows uses the `Makevars` file as well, but I might be wrong.
If that were to be true, then, using the cmake build of `nlopt`, I would need a generated `Makevars` that **always** (no matter which distribution we install **nloptr** on) looks like:

```
CXX_STD = CXX11
PKG_CPPFLAGS = -I../inst/include 
PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) -Lnlopt$(R_ARCH)/lib -lnlopt
```

The difference between `${R_ARCH}` and `\$(R_ARCH)` lies in the use of:
- backslash to escape the `$` operator so that it is not evaluated by the `configure` file but rather copied as is to the `Makevars` file.
- parentheses instead of `{}` because I think environment variables in `Makevars` files are evaluated that way.

But it might be as simple to just add the fixed `Makevars.win` as you just did, in which case you can further simplify the `configure.ac` file I believe this way: `nlopt_libs="-Lnlopt/lib -lnlopt"`.

Reply:
I agree with your second point, sorry about that.

As for the first point, it looks to me that my point stands if you define my way `nlopt_libs` just in the case we have to build nlopt via cmake. And it should avoid the need to use `Makevars.win`. But I might be wrong. You know much better than I do on these topics.

Reply:
Re 2.:  No worries, that was in fact what I tried to modify when I misunderstand what you used `R_ARCH` for _but it simply was never committed_ back to you :)

Re 1:   Yes, sorry, I have to insist here.  This is simply how this work:
- author writes `configure.ac`, tool creates `configure` from it
- `configure` runs and finds replacement values for stub entries
- the stub entries are placed in certain files, in the case notably `src/Makevars.in`
- in general running `configure` converts _any_ files `foo.in` into `foo` (if pointed at by `configure`)
- look at any R package using `configure` and `autoconf`.  Almost all have `src/Makevars.in`, and most also have a (constant, unchanged, as here) `src/Makevars`.

I made another commit that is "marginally more invasive" to bring closure to this PR:  I wanted a run that is all green so I told the Action file to only abort on `error`, not `warning`.  Feel free to revert that after merge but I felt I needed to prove that the PR does what it promises.  I think it does.  If you want I can set up another (trivial) yaml file that does a run _with_ a system nlopt, or if you want you can inject it into the matrix.  It's optional, really.

Reply:
So can we please 'resolve' this subthread, and set it to 'no change required' and merge?

Reply:
Yes I will mark it as resolved. I would be glad if you add another YAML to check build with system nlopt.